### PR TITLE
DHFPROD-5182: Migrate match steps so they are independent of flows

### PR DESCRIPTION
**Pull request by fsnow**
Adding a library to convert matching steps to the new format, a data service that does that conversion and SJS unit tests

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -290,7 +290,7 @@ pipeline{
 		stage('code-review'){
 		when {
   			 allOf {
-    changeRequest author: '', authorDisplayName: '', authorEmail: '', branch: '', fork: '', id: '', target: 'develop', title: '', url: ''
+    changeRequest author: '', authorDisplayName: '', authorEmail: '', branch: '', fork: '', id: '', target: 'feature/5.4-develop', title: '', url: ''
```
Comment (rjrudin) [inline Jenkinsfile:293]: If you do a pull on 5.4-develop locally and then rebase off it, these changes should go away.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.api [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "functionName": "updateOptions",
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.api:2]: I am wondering if we need to distinguish between matching and merging in this function. Also, @ryanjdew - while we prefer matching/merging steps to mastering steps, is it reasonable to still refer to "mastering" as a subsystem, such that it makes sense to have a MasteringService as opposed to separate MatchingService and MergingService interfaces? (I kind of prefer just saying "mastering")

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.sjs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const updateOptionsLib = require("/data-hub/5/data-services/mastering/updateOptionsLib.sjs");
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.sjs:18]: I go back and forth on making a library vs just tossing all the code into the endpoint, and waiting until the need arises to move it into a library for reuse. Thoughts on that? I assume we'll revisit this when we handle merging steps.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.api (hunk 2 above)
Comment (fsnow) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.api:2]: Yes, I will rename this to updateMergeOptions.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.sjs (hunk 3 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/updateOptions.sjs:18]: Personally, I like the library approach. Both match and merge could use the same library and then if we anything other than the data service needs it, we can always move it into another folder.

**Pull request closed without merging**